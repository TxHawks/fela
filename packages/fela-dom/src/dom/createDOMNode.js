/* @flow */
function getDocumentHead(): Object {
  return document.head || {}
}

export default function createDOMNode(
  type: string,
  media: string = '',
  anchorNode: Object
): Object {
  const head = getDocumentHead()

  const node = document.createElement('style')
  node.setAttribute('data-fela-type', type)
  node.type = 'text/css'

  if (media.length > 0) {
    node.media = media
    head.appendChild(node)
  } else {
    // if anchorNode is undefined it will
    // be added at the end by default
    head.insertBefore(node, anchorNode)
  }

  return node
}
