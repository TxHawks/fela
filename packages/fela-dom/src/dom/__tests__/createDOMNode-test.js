import { html as beautify } from 'js-beautify'

import createDOMNode from '../createDOMNode'

describe('Creating DOM node', () => {
  it('should create new DOM nodes', () => {
    createDOMNode('RULE')
    createDOMNode('RULE', '(max-width: 800px)')
    expect(
      beautify(document.documentElement.outerHTML, {
        indent_size: 2
      })
    ).toMatchSnapshot()
  })
})
