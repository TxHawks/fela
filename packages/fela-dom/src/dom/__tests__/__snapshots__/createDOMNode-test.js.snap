// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creating DOM node should create new DOM nodes 1`] = `
"<html>

<head>
  <style data-fela-type=\\"RULE\\" type=\\"text/css\\"></style>
  <style data-fela-type=\\"RULE\\" type=\\"text/css\\" media=\\"(max-width: 800px)\\"></style>
</head>

<body></body>

</html>"
`;
